common/src/main/java/com/wynntils/wc/utils/lootrun/LootrunUtils.java

```diff
@@ -0,0 +1,814 @@
+
+        for (List<ColoredPoint> locationsInRoute : locations) {
+            VertexConsumer consumer = source.getBuffer(LootrunRenderType.LOOTRUN_LINE);
+            Matrix4f lastMatrix = poseStack.last().pose();
+            boolean sourceBatchEnded = false;
+
+            List<ColoredPoint> toRender = new ArrayList<>();
+
+            boolean disablePoint = false;
+            BlockPos lastBlockPos = null;
+
+            for (ColoredPoint point : locationsInRoute) {
+                boolean pauseDraw = false;
+                BlockPos blockPos = new BlockPos(point.vec3());
+
+                if (blockPos.equals(lastBlockPos)) { // Do not recalculate block validness
+                    if (disablePoint) {
+                        pauseDraw = true;
+                    } else if (!toRender.isEmpty()) {
+                        toRender.add(point);
+                    }
+                } else {
+                    Iterable<BlockPos> blocks = getBlocksForPoint(point);
+
+                    boolean barrierInArea = false;
+                    boolean validBlock = false;
+
+                    for (BlockPos blockInArea : blocks) {
+                        BlockState blockStateInArea = level.getBlockState(blockInArea);
+                        if (blockStateInArea.is(Blocks.BARRIER)) {
+                            barrierInArea = true;
+                        } else if (blockStateInArea.getCollisionShape(level, blockInArea) != null) {
+                            validBlock = true;
+                            break;
+                        }
+                    }
+
+                    if (validBlock) {
+                        disablePoint = false;
+                        if (sourceBatchEnded) {
+                            consumer = source.getBuffer(RenderType.lineStrip());
+                            sourceBatchEnded = false;
+                        }
+                        for (ColoredPoint location : toRender) {
+                            Vec3 rawLocation = location.vec3();
+                            int pathColor = location.color();
+                            consumer.vertex(lastMatrix, (float) rawLocation.x, (float) rawLocation.y, (float)
+                                            rawLocation.z)
+                                    .color(pathColor)
+                                    .normal(0, 0, 1)
+                                    .endVertex();
+                        }
+                        toRender.clear();
+                    } else if (barrierInArea) {
+                        disablePoint = true;
+                        pauseDraw = true;
+                        toRender.clear();
+                    } else if (disablePoint) {
+                        pauseDraw = true;
+                    } else {
+                        toRender.add(point);
+                        continue;
+                    }
+                }
+
+                lastBlockPos = blockPos;
+
+                if (!pauseDraw) {
```

Comment:
`pauseDraw` seems to be exactly the same as `disablePoint`, or at least functions the same

Reply:
`pauseDraw` is reset for every point, as it is declared inside the for loop, `disablePoint` is not, it is consistent through rendering.

Reply:
yeah but when `pauseDraw ` is used here its set to the same state as `disablePoint`